lib/Doctrine/Common/Collections/Collection.php

```diff
@@ -41,7 +41,7 @@ interface Collection extends Countable, IteratorAggregate, ArrayAccess
      *
      * @return true Always TRUE.
      */
-    public function add($element): bool;
+    public function add(mixed $element): bool;
```

Comment:
Shall we change the return type to `void`?

Reply:
Yes 👍 Always returning `true` looks weird.

Reply:
I don't think it's a good idea to change return types between 1.x and 2.x. How are we going to support both versions then?

Reply:
Good point! A solution could be to have different traits with different signatures and conditionally use one trait or the other.

Reply:
That will still break the code at large. From our own org we have ODM's dynamic collection generation (which uses trait for generated PersistentCollection so it's ok-ish) but effectively it allowed users to have their own collection classes in documents. I'm myself maintaing https://github.com/malarzm/collections which provides own implementations of `Collection` interface.

My point is: if we require everybody with own implementations of the interface to have traits, adoption will be slow. Always returning `true` may be weird, but bool -> void is too much of a change for a package so widely used.

Reply:
Good point. `Collection` is an interface that should be implemented in userland and the change I suggested would cause too much trouble in existing implementations for too little gain. Let's revert.

Reply:
Reverted. We can deal with this separately, for instance we could drop the return type declaration in 2.0.x, then add `void` in 3.0.x.